ComponentKit/Core/CKComponent.mm

```diff
@@ -164,12 +167,15 @@ - (void)unmount
 
 - (void)_relinquishMountedView
 {
-  UIView *view = _mountInfo->view;
-  if (view) {
-    CKAssert(view.ck_component == self, @"");
-    [_scopeHandle.controller component:self willRelinquishView:view];
-    view.ck_component = nil;
-    _mountInfo->view = nil;
+  CKAssertMainThread();
+  if (_mountInfo != nullptr) {
```

Comment:
If the behavior is consistent elsewhere I'm less concerned about it. I think the main thread assertions are likely enough.